umapi_client/connection.py

```diff
@@ -475,3 +546,84 @@ def call():
         if self.logger: self.logger.error("UMAPI timeout...giving up after %d attempts (%d seconds).",
                                           self.retry_max_attempts, total_time)
         raise UnavailableError(self.retry_max_attempts, total_time, result)
+
+
+class SessionManager:
+    """
+    Handles the session state for the connector.  Allows for variable session life, session
+    regeneration, and the option to disable sessions (connection pooling)
+    """
+    def __init__(self,
+                 user_agent,
+                 connection_pooling,
+                 session_max_age):
+
```

Comment:
By the way, I don't want to argue specific blank lines, but I find blank lines extremely helpful in reading the code as they separate chapters and sections and make structure and function more clear.

Reply:
That's because you've been reading and writing way too much Java code.  They are the silly people who popularized this, because they have so much boilerplate that you need the blank lines in order to read the code at all.  In useful languages such as Python, people are aware of the drawbacks of blank lines: since there is no agreed notion of "chapter" and "section", blank lines just make the code incredibly unmaintainable over time.  If there really is a section of code that needs to be separated and explained, then have a block comment at the top of the section and another at the bottom.  Here's a [good blog post by a Java guy](https://www.yegor256.com/2014/11/03/empty-line-code-smell.html) that I think gets right to the point.